### PR TITLE
feat: Update nav bar with new data information architecture for SQL Lab + SQL

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Updating the data information architecture for the nav bar to match the new designs from product. Mostly renaming SQL LAB -> SQL + SQL Editor to SQL LAB. This PR doesn't change the underlying permissions but does, update the labels and layout there will be a follow up PR once for renaming once we are getting closer to 3.0 cut

Screen Shot 2022-09-02 at 11 16 03 AM

Screen Shot 2022-09-02 at 11 16 11 AM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by betodealmeida** [comment]
Is this changing the UI? Can you add a summary and screenshots as well if it's the case?

**Review by AAfghahi — APPROVED** [review]

**Review by Antonio-RiveroMartnez — APPROVED** [review]
LGTM!

File: superset/initialization/__init__.py [hunk 1]
```diff
@@ -328,18 +348,19 @@ def init_views(self) -> None:
...
         appbuilder.add_link(
             "SQL Editor",
-            label=_("SQL Editor"),
+            label=_("SQL Lab"),
             href="/superset/sqllab/",
             category_icon="fa-flask",
             icon="fa-flask",
             category="SQL Lab",
-            category_label=__("SQL Lab"),
+            category_label=__("SQL"),
```
Comment (eschutho) [inline superset/initialization/__init__.py:356]: is this currently using the menu permissions or the can sql lab permissions?

Comment (hughhhh) [reply superset/initialization/__init__.py:356]: it's using the can_sqllab permission

**Review by dpgaspar — COMMENTED** [review]
can you include your change on https://github.com/apache/superset/pull/21297 into this PR?

**Pull request merged**